lib/commands/exam.js

```diff
@@ -124,35 +180,98 @@ module.exports = TestCommand.extend({
    * @override
    */
   _generateCustomConfigs: function(commandOptions) {
-    var config = this._super._generateCustomConfigs.apply(this, arguments);
-
-    if (this.validator.shouldParallelize) {
-      // Get the testPage from the generated config or the Testem config and
-      // use it as the baseUrl for the parallelized test pages
-      var baseUrl = config.testPage || this._getTestPage(commandOptions.configFile);
-      var count = commandOptions.split;
-      var partitions = commandOptions.partition;
-
-      if (!partitions) {
-        partitions = [];
-        for (var i = 0; i < commandOptions.split; i++) {
-          partitions.push(i + 1);
-        }
-      }
+    const config = this._super._generateCustomConfigs.apply(this, arguments);
+    const customBaseUrl = this._getCustomBaseUrl(config, commandOptions);
 
-      if (Array.isArray(baseUrl)) {
-        var command = this;
-        config.testPage = baseUrl.reduce(function(testPages, baseUrl) {
-          return testPages.concat(command._generateTestPages(baseUrl, partitions, count));
-        }, []);
-      } else {
-        config.testPage = this._generateTestPages(baseUrl, partitions, count);
+    const shouldParallelize = this.validator.shouldParallelize;
+    const shouldLoadBalance = this.validator.shouldLoadBalance;
+    const shouldReplayExecution = this.validator.shouldReplayExecution;
+
+    if (!shouldParallelize && !shouldLoadBalance && !shouldReplayExecution) {
+      return config;
+    }
+
+    let browserIds = commandOptions.partition;
+    let appendingParam = '_partition';
+
+    if (shouldParallelize && !browserIds) {
+      browserIds = [];
+      for (var i = 0; i < commandOptions.split; i++) {
+        browserIds.push(i + 1);
       }
+    } else if (shouldLoadBalance) {
+      config.additional_browser_socket_events = this._addLoadBalancingBrowserSocketEvents(config);
+
+      const browserCount = commandOptions.loadBalance;
+      appendingParam = '_browser';
+
+      // Creates an array starting from 1 to browserCount. e.g. if brownserCount is 3, the returned array is [1, 2, 3]
+      browserIds = Array.from({length: browserCount}, (v,k) => k+1);
+    } else if (shouldReplayExecution) {
+      const executionMapping = this.getExecutionMappigInstance(commandOptions);
+      config.browser_module_mapping = executionMapping['executionMapping'];
+
```

Comment:
We will also need 
`config.additional_browser_socket_events = this._addLoadBalancingBrowserSocketEvents(config);`
in this if condition, since it still requires the socket events to be setup